File: flutter_laravel/laravel/app/Models/Subscription.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Subscription extends Model
{
    protected $fillable = [
        'user_id', 'merchant_name', 'amount', 'currency',
        'billing_cycle', 'next_renewal_date', 'category',
        'status', 'confidence_score', 'bank_connection_id',
    ];

    protected $casts = [
        'amount' => 'decimal:2',
        'confidence_score' => 'float',
        'next_renewal_date' => 'date',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function cancellations(): HasMany
    {
        return $this->hasMany(CancellationRequest::class);
    }
}

File: flutter_laravel/flutter/lib/core/models/subscription.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'subscription.freezed.dart';
part 'subscription.g.dart';

@freezed
class Subscription with _$Subscription {
  const factory Subscription({
    required int id,
    required int userId,
    required String merchantName,
    required double amount,
    required String currency,
    required String billingCycle,
    String? nextRenewalDate,
    String? category,
    required String status,
    required double confidenceScore,
    int? bankConnectionId,
    required String createdAt,
    required String updatedAt,
  }) = _Subscription;

  factory Subscription.fromJson(Map<String, dynamic> json) =>
      _$SubscriptionFromJson(json);
}
